x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlInitializationService.java

```diff
@@ -112,5 +126,33 @@ MlDailyMaintenanceService getDailyMaintenanceService() {
         return mlDailyMaintenanceService;
     }
 
+    private void makeMlIndicesHidden() {
+        String[] mlHiddenIndexPatterns = MachineLearning.getMlHiddenIndexPatterns();
+        GetSettingsResponse getSettingsResponse =
+            client.admin().indices().prepareGetSettings()
+                .setIndices(mlHiddenIndexPatterns)
+                .setIndicesOptions(IndicesOptions.LENIENT_EXPAND_OPEN_CLOSED_HIDDEN)
+                .get();
+        String[] nonHiddenIndices =
+            getSettingsResponse.getIndexToSettings().stream()
+                .filter(e -> e.getValue().getAsBoolean(SETTING_INDEX_HIDDEN, false) == false)
+                .map(Map.Entry::getKey)
+                .toArray(String[]::new);
+        if (nonHiddenIndices.length == 0) {
+            logger.info("There are no indices that need to be made hidden, " + getSettingsResponse);
+            return;
+        }
+        String nonHiddenIndicesString = Arrays.stream(nonHiddenIndices).collect(Collectors.joining(", "));
+        logger.info("The following indices will now be made hidden: {}", nonHiddenIndicesString);
+        AcknowledgedResponse updateSettingsResponse =
+            client.admin().indices().prepareUpdateSettings()
+                .setIndicesOptions(IndicesOptions.LENIENT_EXPAND_OPEN_CLOSED_HIDDEN)
+                .setIndices(nonHiddenIndices)
+                .setSettings(Collections.singletonMap(SETTING_INDEX_HIDDEN, true))
```

Comment:
Do we also need to make the aliases hidden?

Reply:
Done.